integration_tests/commands/toggle_test.go

```diff
@@ -0,0 +1,71 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/dicedb/dice/testutils"
+	"gotest.tools/v3/assert"
+)
+
+func TestEvalJSONTOGGLE(t *testing.T) {
+    conn := getLocalConnection()
+    defer conn.Close()
+
+	simpleJSON :=`{"name":true,"age":false}`
+	complexJson :=`{"field":true,"nested":{"field":false,"nested":{"field":true}}}`
+
+    testCases := []struct {
+        name     string
+        commands []string
+        expected []interface{}
+    }{
+        {
+			name:     "JSON.TOGGLE with existing key",
+			commands: []string{`JSON.SET user $ `+ simpleJSON, "JSON.TOGGLE user $.name"},
+			expected: []interface{}{"OK", []any{int64(0)}},
+		},
+        {
+            name:     "JSON.TOGGLE with non-existing key",
+            commands: []string{"JSON.TOGGLE user $.flag"},
+            expected: []interface{}{"ERR could not perform this operation on a key that doesn't exist"},
+        },
+        {
+			name:     "JSON.TOGGLE with invalid path",
+			commands: []string{`JSON.SET testkey $ ` + simpleJSON, "JSON.TOGGLE user $.invalidPath"},
+			expected: []interface{}{"WRONGTYPE Operation against a key holding the wrong kind of value", "ERR could not perform this operation on a key that doesn't exist"},
+		},
+        {
+            name:     "JSON.TOGGLE with invalid command format",
+            commands: []string{"JSON.TOGGLE testKey"},
+            expected: []interface{}{"ERR wrong number of arguments for 'json.toggle' command"},
+        },
+		{
+            name:     "deeply nested JSON structure with multiple matching fields",
+            commands: []string{
+                `JSON.SET user $ `+ complexJson,
+				"JSON.GET user",
+				"JSON.TOGGLE user $..field",
+				"JSON.GET user",
+            },
+            expected: []interface{}{"OK",
+			"{\"field\":true,\"nested\":{\"field\":false,\"nested\":{\"field\":true}}}",
+			[]any{int64(0), int64(1), int64(0)},
+			"{\"field\":false,\"nested\":{\"field\":true,\"nested\":{\"field\":false}}}"},
+        },
+    }
+
+    for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			FireCommand(conn, "DEL user") 
+	
+			for i, cmd := range tc.commands {
+				result := FireCommand(conn, cmd)
+				if expectedSlice, ok := tc.expected[i].([]interface{}); ok {
+					assert.Assert(t, testutils.UnorderedEqual(expectedSlice, result))
+				} else {
+					assert.DeepEqual(t, tc.expected[i], result)
```

Comment:
You are directly comparing the string representations of the json objects here, the field ordering for json elements is not deterministic - which means the test is flaky om the CI (you can see the failure in the checks).
Unmarshal the expected and actual values into a map and do a deep equal on those.

Reply:
You are correct. Sometimes the tests are passing but sometimes they are failing due to non-deterministic nature of JSON strings so i unmarshal them and stored them in the the map and then compared.